controller/internal/recorder/event.go

```diff
@@ -0,0 +1,22 @@
+package recorder
+
+import (
+	"context"
+
+	"github.com/giantswarm/microerror"
+	corev1 "k8s.io/api/core/v1"
+	pkgruntime "k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/client-go/tools/record"
+)
+
+type Recorder struct {
```

Comment:
Some opinions. 

* Let's rename `service.go` to `recorder.go`. That service naming is not good as we learned in the past.
* Let's move `Recorder` to `recorder.go`. The struct and constructor magic should be in the same file. 
* Let's move the `Emit` method to `recorder.go`. No reason to keep 7 lines in a separate file.